### ISSUE TITLE
Compiling with new RNNoise on Linux

### ISSUE THREAD

**Issue by LiamDawe** [obsproject/obs-studio#3352]
Hi, I am attempting to test the new RC, which includes the newer RNNoise filter for noise suppression. Does this need new dependencies? I didn't see any that mentioned it in the docs, so do they perhaps need adjusting for this new feature?

As when going to compile it doesn't find it:

> -- Could NOT find Librnnoise (missing: RNNOISE_LIB RNNOISE_INCLUDE_DIR) 

I have installed "[librnnoise-ladspa](https://aur.archlinux.org/packages/librnnoise-ladspa/)" from the Arch AUR, which does give me the /usr/lib/librnnoise_ladspa.so file needed for it in PulseAudio. So I am curious as to how I can get this working to test?

**Comment by jp9000** [comment]
Hm, it's possible that the `_ladspa` part is messing with its detection.  Not entirely sure.

**Comment by Gol-D-Ace** [comment]
This looks like a plug-in and not the library itself which is needed for OBS.

**Comment by LGCW** [comment]
I would expect it to look for librnnoise_vst.so in ~/.vst. 

It does not.

**Comment by jp9000** [comment]
You could try manually building it in and setting the library path in cmake settings to see if it works.  If it is the right library, maybe I could change some cmake config around to make it search for that name in addition to the expected name

**Comment by LiamDawe** [comment]
@Gol-D-Ace yes, it's the Pulse plugin. If that's not what's needed, can we get some instructions sorted somewhere on what is actually needed for that feature? Anyone know the best way on Linux for it? Happy to help test if someone can point me in the right direction.

**Comment by WizardCM** [comment]
@LiamDawe you can find the repo & build instructions here https://github.com/xiph/rnnoise

**Comment by johnpyp** [comment]
Additionally, the default place where the rnnoise makefile puts the library is in `/usr/local/lib`, but, at least on my arch system, `/usr/local/lib` wasn't in LD_LIBRARY_PATH by default, so I moved it to `/usr/lib` which is in that path. When obs starts it will launch, but fail to initialize the filters stuff because librnnoise isn't dynamically linked properly (since it couldn't be found in the LD_LIBRARY_PATH).

Basically, while building OBS the normal rnnoise build process worked fine for me because OBS will explicitly look in `/usr/local/lib`, but at runtime it won't work unless `/usr/local/lib` is in the LD_LIBRARY_PATH.
